packages/swapper/src/swappers/RelaySwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,242 @@
+import type { Execute } from '@reservoir0x/relay-sdk'
+import {
+  bnOrZero,
+  convertBasisPointsToPercentage,
+  convertDecimalPercentageToBasisPoints,
+  isSome,
+} from '@shapeshiftoss/utils'
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+import type { AxiosResponse } from 'axios'
+
+import type {
+  CommonTradeQuoteInput,
+  GetTradeRateInput,
+  SwapErrorRight,
+  SwapperConfig,
+  SwapperDeps,
+  TradeQuoteStep,
+} from '../../../types'
+import { MixPanelEvent, SwapperName, TradeQuoteError } from '../../../types'
+import { makeSwapErrorRight } from '../../../utils'
+import type { relayChainMap as relayChainMapImplementation } from '../constant'
+import { getRelayEvmAssetAddress } from '../utils/getRelayEvmAssetAddress'
+import { relayService } from '../utils/relayService'
+import { relayTokenToAsset } from '../utils/relayTokenToAsset'
+import { relayTokenToAssetId } from '../utils/relayTokenToAssetId'
+import type { QuoteParams, RelayTradeQuote, RelayTradeRate } from '../utils/types'
+import { isRelayToken } from '../utils/types'
+
+// @TODO: implement affiliate fees
+export const getQuote = async (
+  params: QuoteParams,
+  config: SwapperConfig,
+): Promise<Result<AxiosResponse<Execute, any>, SwapErrorRight>> => {
+  return await relayService.post<Execute>(`${config.VITE_RELAY_API_URL}/quote`, params)
+}
+
+export async function getTrade({
+  input,
+  deps,
+  relayChainMap,
+}: {
+  input: CommonTradeQuoteInput | GetTradeRateInput
+  deps: SwapperDeps
+  relayChainMap: typeof relayChainMapImplementation
+}): Promise<Result<RelayTradeQuote[] | RelayTradeRate[], SwapErrorRight>> {
+  const {
+    sellAsset,
+    buyAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit,
+    sendAddress,
+    receiveAddress,
+    accountNumber,
+    affiliateBps,
+    potentialAffiliateBps,
+    slippageTolerancePercentageDecimal: _slippageTolerancePercentageDecimal,
+  } = input
+
+  const slippageToleranceBps = _slippageTolerancePercentageDecimal
+    ? convertDecimalPercentageToBasisPoints(_slippageTolerancePercentageDecimal).toFixed()
+    : undefined
+
+  const sellRelayChainId = relayChainMap[sellAsset.chainId]
+  const buyRelayChainId = relayChainMap[buyAsset.chainId]
+
+  if (sellRelayChainId === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${sellAsset.name}' on chainId '${sellAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  if (buyRelayChainId === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${buyAsset.name}' on chainId '${buyAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  if (!sendAddress) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'sendAddress is required',
+      }),
+    )
+  }
+
+  const maybeQuote = await getQuote(
```

Comment:
suggestion: this is quite confusing to grasp (`getTrade` / `getQuote`).

Can we rename `getQuote` to `fetchRelayQuote` / `fetchRelayTrade` (I actually like the former more, as this decouples from our quote/rates vernacular) to signify fetching intent

Reply:
Done in [7e3a115](https://github.com/shapeshift/web/pull/9188/commits/7e3a115c7aeb37843ca13844d6f4f73501a40f03) , I'm agnostic on this one